### PR TITLE
PSVAMB-19043 Port the plugin from V2

### PR DESCRIPTION
**Pull request by yusitnikov**
This PR contains the minimal code that is required for the plugin to work.
All nice extras (docs, lint, typescript etc.) will come in a separate PR.

Original plugin code: https://github.com/kaltura/kwidget-ps/blob/master/ps/Vision/resources/visibleWatermark.js

### CODE REVIEW

**Review by OrenMe — COMMENTED** [review]
great work.
Following our call, non essential suggestions for improvement:
1. add tests
2. readme
3. eslint and prettier

File: src/components/WatermarkContainer/WatermarkContainer.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    return (
+        isVisible && <div className={`${styles.container} ${styles['position' + position]}`}>
+            {text}
+        </div>
+    );
+};
+
+// Monitor the player state
+const mapStateToProps = (state) => ({isPlaying: !!state.engine.playerState.currentState});
```
Comment (yairans) [inline src/components/WatermarkContainer/WatermarkContainer.js]: `!!state.engine.playerState.currentState` should be always `true`
maybe `state.engine.isPlaying` and `state.engine.adIsPlaying` will be useful for you

File: src/watermark.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        duration: 2,
+        text: 'watermarked',
+        showEvery: 120
+    };
+
+    static isValid() {
+        return true;
+    }
+
+    getUIComponents() {
```
Comment (yairans) [inline src/watermark.js:15]: important to know that now the player supports adding (and replacing/removing) component  dynamically
by `player.ui.addComponent` api (gets the same object as `getUIComponents` )

**Review by yairans — CHANGES_REQUESTED** [review]

**Comment by yairans** [comment]
@yusitnikov great plugin, please see my comments.

File: src/components/WatermarkContainer/WatermarkContainer.js (hunk 1 above)
Comment (yusitnikov) [reply src/components/WatermarkContainer/WatermarkContainer.js]: I'm trying to catch the "welcome" state - when the player was just rendered, but the user didn't press "play". I see that `state.engine.playerState.currentState` is an empty string in this state. Is it going to be changed?

File: src/watermark.js (hunk 2 above)
Comment (yusitnikov) [reply src/watermark.js:15]: I know, thanks :)

File: src/components/WatermarkContainer/WatermarkContainer.js (hunk 1 above)
Comment (yusitnikov) [reply src/components/WatermarkContainer/WatermarkContainer.js]: nm, found the relevant flag :)

**Pull request merged**